random-beans/src/test/java/io/github/benas/randombeans/PopulatorImplTest.java

```diff
@@ -145,6 +146,23 @@ public void javaNetTypesShouldBePopulated() throws Exception {
         assertThat(website.getUrl()).isNotNull();
     }
 
+    @Test(expected = BeanPopulationException.class)
+    public void failsToPopulateInterfacesAndAbstractClassesIfScanClasspathForConcreteClassesIsDisabled() throws Exception {
+        Populator nonScanningPopulator = aNewPopulatorBuilder().scanClasspathForConcreteClasses(false).build();
+
+        nonScanningPopulator.populateBean(Mamals.class);
+    }
+
+    @Test
+    public void generatesConcreteTypesForInterfacesAndAbstractClassesIfScanClasspathForConcreteClassesIsEnabled() throws Exception {
+        Populator scanningPopulator = aNewPopulatorBuilder().scanClasspathForConcreteClasses(true).build();
+
+        Mamals mamals = scanningPopulator.populateBean(Mamals.class);
+
+        assertThat(mamals.getMamal()).isOfAnyClassIn(Human.class, Ape.class, Person.class, SocialPerson.class);
+        assertThat(mamals.getMamalImpl()).isOfAnyClassIn(Human.class, Ape.class, Person.class, SocialPerson.class);
+    }
+
```

Comment:
According to http://tutorials.jenkov.com/java-reflection/generics.html you need the field to determine the generic type of the `Comparable`. Sadly in `doPopulateBean()` you do not have access to the field type anymore. So it is not possible choose a subclasses which implements  `Comparable<Date>`. The current implementation just chooses a random `Comparable`.